### PR TITLE
Rehome python bazel distribtest

### PR DESCRIPTION
**Pull request by gnossen**
This moves all existing Bazel distribtests to the same directory.

I also needed to run Python formatting since the previous directory was apparently excluded from formatting.

### CODE REVIEW

**Review by jtattermusch — COMMENTED** [review]
I think overall this looks good. I'd defer to @lidizheng  for review of the changes to the python distribtest code.

Also, currently the tests are failing.

**Review by lidizheng — COMMENTED** [review]
Looks good. The distribtest for Bazel is failing due to Python dependency issue, we might want to double check.

(I confused by the name a bit, it's testing Bazel versions, instead of using Bazel to test Python or C++.)

File: test/distrib/bazel/test_single_bazel_version.sh [hunk 1]
```diff
@@ -41,16 +41,25 @@ EXCLUDED_TARGETS=(
...
 export OVERRIDE_BAZEL_VERSION="$VERSION"
 # when running under bazel docker image, the workspace is read only.
 export OVERRIDE_BAZEL_WRAPPER_DOWNLOAD_DIR=/tmp
 bazel build -- //... "${EXCLUDED_TARGETS[@]}" || FAILED_TESTS="${FAILED_TESTS}Build "
 
-cd test/distrib/bazel/cpp/
+for TEST_DIRECTORY in "${TEST_DIRECTORIES[@]}"; do
+  pushd "test/distrib/bazel/$TEST_DIRECTORY/"
+
+  bazel test --test_output=errors //:all || FAILED_TESTS="${FAILED_TESTS}${TEST_DIRECTORY} Distribtest"
```
[the remark this answers is not in the document]
Comment (gnossen) [reply test/distrib/bazel/test_single_bazel_version.sh]: Done.

File: test/distrib/bazel/python/namespaced/upper/example/no_import_strip_test.py [hunk 2]
```diff
@@ -17,15 +17,17 @@
...
         namespaced_example.value = "hello"
-        # Dummy assert, important part is namespaced example was imported.
+        # Superfluous assert, important part is namespaced example was imported.
         self.assertEqual(namespaced_example.value, "hello")
 
     def test_grpc(self):
         from namespaced_example_pb2_grpc import NamespacedServiceStub
+
         # No error from import
         self.assertEqual(1, 1)
```
[the remark this answers is not in the document]
Comment (gnossen) [reply test/distrib/bazel/python/namespaced/upper/example/no_import_strip_test.py:32]: No, not that I'm aware of.

**Comment by gnossen** [comment]
Ah, looks like `isort` broke the test. Interesting. I don't see why this should have failed. I guess this was a latent bug. I've filed [a TODO](https://github.com/grpc/grpc/issues/29284) for this to tackle fixing the latent bug, but just turned off `isort` in this block for now.

> (I confused by the name a bit, it's testing Bazel versions, instead of using Bazel to test Python or C++.)

The point of the Bazel distribtest is to ensure that people who pull in gRPC as a dependency via Bazel are able to use it properly. Specifically, this tests:

- `grpc_deps`
- `grpc_extra_deps`
- `py_proto_library`
- `py_grpc_library`
- `cc_grpc_library`

And various features of all of the above. The Python Bazel distribtest in particular has really good feature coverage because of community contributions.

**Review by lidizheng — APPROVED** [review]

**Pull request merged**